espnet/nets/pytorch_backend/e2e_st_conformer.py

```diff
@@ -49,6 +51,21 @@ def __init__(self, idim, odim, args, ignore_id=-1):
         super().__init__(idim, odim, args, ignore_id)
         if args.transformer_attn_dropout_rate is None:
             args.transformer_attn_dropout_rate = args.dropout_rate
+
+        # Check the relative positional encoding type
+        self.rel_pos_type = getattr(args, "rel_pos_type", None)
+        if self.rel_pos_type is None or self.rel_pos_type == "legacy":
+            if args.transformer_encoder_pos_enc_layer_type == "rel_pos":
+                args.transformer_encoder_pos_enc_layer_type = "legacy_rel_pos"
+                logging.warning(
+                    "Using legacy_rel_pos and it will be deprecated in the future."
+                )
+            if args.transformer_encoder_selfattn_layer_type == "rel_selfattn":
+                args.transformer_encoder_selfattn_layer_type = "legacy_rel_selfattn"
+                logging.warning(
+                    "Using legacy_rel_selfattn and it will be deprecated in the future."
+                )
+
```

Comment:
```suggestion
        args = verify_rel_pos_type(args)
```

Reply:
Then, you can define `verify_rel_pos_type` in `augment.py` as follows:
```
def verify_rel_pos_type(args):
    """Verify the relative positional encoding type for compatibility.

    Args:
        args (Namespace): original arguments
    Returns:
        args (Namespace): modified arguments

    """
    rel_pos_type = getattr(args, "rel_pos_type", None)
    if rel_pos_type is None or rel_pos_type == "legacy":
        if args.transformer_encoder_pos_enc_layer_type == "rel_pos":
            args.transformer_encoder_pos_enc_layer_type = "legacy_rel_pos"
            logging.warning(
                "Using legacy_rel_pos and it will be deprecated in the future."
            )
        if args.transformer_encoder_selfattn_layer_type == "rel_selfattn":
            args.transformer_encoder_selfattn_layer_type = "legacy_rel_selfattn"
            logging.warning(
                "Using legacy_rel_selfattn and it will be deprecated in the future."
            )
    return args
```

Reply:
@pengchengguo Could you do this in Transformer ASR as well?

Reply:
Ok! It will be more concise.